lgc/builder/YCbCrConverter.cpp

```diff
@@ -326,64 +328,78 @@ void YCbCrConverter::genSamplerDescChroma() {
+    dstSelXYZW = m_builder->CreateSelect(isBGRGFmt, m_builder->getInt32(0xF2E), dstSelXYZW);
 
-  switch (m_metaData.word1.planes) {
-  case 1:
-    proxySqRsrcRegHelper.setReg(SqRsrcRegs::DstSelXYZW, m_builder->getInt32(m_metaData.word1.dstSelXYZW));
-    proxySqRsrcRegHelper.setReg(SqRsrcRegs::BcSwizzle, m_builder->getInt32(6));
+    YCbCrAddressHandler addrHelper(m_builder, &proxySqRsrcRegHelper, m_gfxIp);
+    addrHelper.genHeightAndPitch(m_metaData.word0.bitDepth.channelBitsR, 32, m_metaData.word2.bitCounts.xBitCount,
+                                 m_metaData.word1.tileOptimal, m_metaData.word1.planes);
+    proxySqRsrcRegHelper.setReg(SqRsrcRegs::Width,
+                                m_builder->CreateLShr(width, ConstantInt::get(m_builder->getInt32Ty(), 1)));
+    proxySqRsrcRegHelper.setReg(SqRsrcRegs::DstSelXYZW, dstSelXYZW);
     proxySqRsrcRegHelper.setReg(SqRsrcRegs::Pitch, addrHelper.getPitchCb());
     m_imgDescsChroma[1] = proxySqRsrcRegHelper.getRegister();
-    break;
-  case 2:
-    proxySqRsrcRegHelper.setReg(SqRsrcRegs::BaseAddress, addrHelper.getPlane(1));
-    proxySqRsrcRegHelper.setReg(SqRsrcRegs::Height, m_metaData.word1.ySubSampled ? heightHalf : height);
-    proxySqRsrcRegHelper.setReg(SqRsrcRegs::DstSelXYZW, m_builder->getInt32(m_metaData.word1.dstSelXYZW));
-    proxySqRsrcRegHelper.setReg(SqRsrcRegs::BcSwizzle, m_builder->getInt32(6));
-    proxySqRsrcRegHelper.setReg(SqRsrcRegs::Pitch, pitchHalf);
-    m_imgDescsChroma[1] = proxySqRsrcRegHelper.getRegister();
-    break;
-  case 3:
-    proxySqRsrcRegHelper.setReg(SqRsrcRegs::BaseAddress, addrHelper.getPlane(1));
-    proxySqRsrcRegHelper.setReg(SqRsrcRegs::Height, heightHalf);
-    proxySqRsrcRegHelper.setReg(SqRsrcRegs::DstSelXYZW, m_builder->getInt32(0x300));
-    proxySqRsrcRegHelper.setReg(SqRsrcRegs::BcSwizzle, m_builder->getInt32(4));
-    proxySqRsrcRegHelper.setReg(SqRsrcRegs::Pitch, pitchHalf);
-    m_imgDescsChroma[1] = proxySqRsrcRegHelper.getRegister();
-
-    proxySqRsrcRegHelper.setReg(SqRsrcRegs::BaseAddress, addrHelper.getPlane(2));
-    proxySqRsrcRegHelper.setReg(SqRsrcRegs::DstSelXYZW, m_builder->getInt32(0x204));
-    proxySqRsrcRegHelper.setReg(SqRsrcRegs::BcSwizzle, m_builder->getInt32(5));
-    m_imgDescsChroma[2] = proxySqRsrcRegHelper.getRegister();
-    break;
-  default:
-    llvm_unreachable("Out of range plane count!");
-    break;
   }
 }
 
+// =====================================================================================================================
+// Set image descriptor for chroma channel
+//
+// @param imageDescArray : Image descriptor array (Up to 3 descriptor is allowed)
+// @param planeIndex : Specific plane index for setting image descriptor
+void YCbCrConverter::SetImgDescChroma(llvm::Value *imageDescArray, unsigned planeIndex) {
+  assert(planeIndex < 3);
+  m_imgDescsChroma[planeIndex] = m_builder->CreateExtractValue(imageDescArray, planeIndex);
+}
```

Comment:
The interface of this method kind of conflates two different things:

1. Extracting a descriptor from an array of descriptors.
2. Setting that descriptor as the plane's image descriptor.

Please change the interface to be:
```
YCbCrConverter::SetImgDescChroma(unsigned planeIndex, llvm::Value *imageDesc);
```
instead. This more clearly captures that we're assigning a single image descriptor to a given plane index, and makes the method more aligned with `GetImgDescChroma`.